### PR TITLE
Default language set when no selection done

### PR DESCRIPTION
**Pull request by peter-ochieng**
# Description

Selecting no language should not show translation keys but instead translated using in english

medic/medic#5432

# Review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- [ ] Documented: Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: webapp/src/js/controllers/user-language-modal.js [hunk 1]
```diff
@@ -44,10 +44,8 @@ angular.module('inboxControllers').controller('UserLanguageModalCtrl',
     };
 
     $scope.submit = function() {
-      if (!$scope.selectedLanguage) {
-        var err = new Error('No language selected');
-        $log.error(err);
-        return $q.reject(err);
+      if (typeof $scope.selectedLanguage === 'undefined') {
```
Comment (garethbowen) [inline webapp/src/js/controllers/user-language-modal.js]: What if (somehow) it's `null` or `""` or something? I feel like the original code `!$scope.selectedLanguage` is a safer option here.

File: webapp/src/js/controllers/user-language-modal.js [hunk 2]
```diff
@@ -44,10 +44,8 @@ angular.module('inboxControllers').controller('UserLanguageModalCtrl',
...
+        $scope.changeLanguage('en');
```
Comment (garethbowen) [inline webapp/src/js/controllers/user-language-modal.js]: This should default to the application default language not 'en'. Use the Settings service and get the `locale` property.

Also, if they're submitting it we should update the user so they don't get prompted next time, so call UpdateUser with the app default.

File: webapp/src/js/controllers/user-language-modal.js [hunk 3]
```diff
@@ -64,8 +62,13 @@ angular.module('inboxControllers').controller('UserLanguageModalCtrl',
 
     $scope.cancel = function() {
       // Reset to initial language.
-      $scope.changeLanguage(initialLanguageCode);
       $uibModalInstance.dismiss();
     };
+
+    $scope.$on('modal.closing', function() {
+      if ($scope.selectedLanguage.toString() === 'undefined') {
+        $scope.changeLanguage('en');
```
Comment (garethbowen) [inline webapp/src/js/controllers/user-language-modal.js]: As above - use the app default, not 'en'.

**Comment by garethbowen** [comment]
@peter-ochieng The merge build is failing due to some new harsher eslint rules, and the other build failed on karma tests. Fix those and you're good to go!

**Comment by newtewt** [comment]
@garethbowen  this was pretty simple to reproduce. Pressing submit without selecting anything forced the undefined value.

**Comment by garethbowen** [comment]
@newtewt Doesn't work for me. The difference may be how you created the user. Can you outline the steps here? Was it through the UI or using medic-conf CLI or other? Did you specify a language? etc etc...

**Comment by newtewt** [comment]
It was via the admin console through the webapp.  Attached is a picture of the info filled in during creating sans password. 

image

**Comment by garethbowen** [comment]
@newtewt Thanks. Can you please try one more thing? If you submit that page, but before you log in as the new user, go to `http://<server>/medic/org.couchdb.user:<username>`. I'd be really interested in the JSON you get!

**Comment by newtewt** [comment]
```
{
  "_id": "org.couchdb.user:tedd",
  "_rev": "1-1b96d880d75b7aa1f8821eda9387c83d",
  "name": "tedd",
  "type": "user-settings",
  "roles": ["district_admin"],
  "facility_id": "b9d543cf-1c35-468a-9ab7-f183bfae2e44",
  "contact_id": "67fdec5d-ebaf-4311-99b7-c87f9d1dfc1b"
}

```

**Comment by newtewt** [comment]
and after logging in and not selecting a language. 

```
{
  "_id": "org.couchdb.user:tedd",
  "_rev": "3-d096da0a88f2f473f1815c2ac2f87d77",
  "name": "tedd",
  "type": "user-settings",
  "roles": ["district_admin"],
  "facility_id": "b9d543cf-1c35-468a-9ab7-f183bfae2e44",
  "contact_id": "67fdec5d-ebaf-4311-99b7-c87f9d1dfc1b",
  "language": "undefined",
  "known": true
}

```

**Comment by garethbowen** [comment]
Thanks! So the user creation looks fine, it's the modal (or somewhere else in the login sequence) that's buggy.

**Comment by garethbowen** [comment]
Closing - this has been replaced by a different solution here: #5496

**Pull request closed without merging**